instrumentation/runtime-telemetry/runtime-telemetry-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java8/Threads.java

```diff
@@ -116,6 +116,12 @@ private static boolean isJava9OrNewer() {
     return THREAD_INFO_IS_DAEMON != null;
   }
 
+  private static boolean useThreads() {
+    // GraalVM native image does not support ThreadMXBean yet
+    // see https://github.com/oracle/graal/issues/6101
+    return !isJava9OrNewer() || System.getProperty("org.graalvm.nativeimage.imagecode") != null;
```

Comment:
I think that GraalVM Native Image would work fine with  `Threads::java8Callback`. Native Image implements some `ThreadMXBean` functionality, notably all the functionality needed by `Threads::java8Callback` (see [here](https://github.com/oracle/graal/blob/jdk-25%2B7/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/management/SubstrateThreadMXBean.java#L149) and [here](https://github.com/oracle/graal/blob/jdk-25%2B7/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/management/SubstrateThreadMXBean.java#L126)). `java9AndNewerCallback` still won't work with Native Image since we don't support `ThreadMXBean#getAllThreadIds()` or `ThreadMXBean.getThreadInfo()` yet.

Reply:
I tried JMX - and ran into https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13078#issuecomment-2616399563